biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -131,10 +138,29 @@ private Repository readRepository() throws Exception {
 		List<Artifact> artifacts = p2.getArtifacts();
 		List<Promise<Resource>> fetched = new ArrayList<>(artifacts.size());
 		Set<URI> visited = new HashSet<>(artifacts.size());
+		Set<String> visitedHash = new HashSet<>(artifacts.size());
+		Map<String,Resource> existing = new HashMap<>();
+
+		if (getBridge() != null) {
+			Requirement requirement = new RequirementBuilder(P2_CAPABILITY_NAMESPACE).addAttribute(MD5_ATTRIBUTE, "*")
```

Comment:
This requirement is static. It does not change upon each invocation. So why not move it to a static final field? In fact the static final field should be: 
```
private static final List<Requirement> hasMD5 = Collections.singletonList(new RequirementBuilder(P2_CAPABILITY_NAMESPACE).addAttribute(MD5_ATTRIBUTE, "*").buildSyntheticRequirement());
```